omnigibson/simulator.py

```diff
@@ -589,32 +509,26 @@ def n_physics_timesteps_per_render(self):
         assert n_physics_timesteps_per_render.is_integer(), "render_timestep must be a multiple of physics_timestep"
         return int(n_physics_timesteps_per_render)
 
-    def step(self, render=True, force_playing=False):
+    def step(self, render=True):
         """
         Step the simulation at self.render_timestep
 
         Args:
             render (bool): Whether rendering should occur or not
-            force_playing (bool): If True, will force physics to propagate (i.e.: set simulation, if paused / stopped,
-                to "play" mode)
         """
-        # Possibly force playing
-        if force_playing and not self.is_playing():
-            self.play()
+        # If we have imported any objects within the last timestep, we render the app once, since otherwise calling
+        # step() may not step physics
+        if len(self._objects_to_initialize) > 0:
+            self.render()
```

Comment:
Maybe we should add a TODO here to check with Nvidia.

Reply:
added doc we can add to